Zend/zend_language_scanner.l

```diff
@@ -2368,6 +2372,10 @@ inline_char_handler:
 	RETURN_TOKEN_WITH_STR(T_NAME_RELATIVE, sizeof("namespace\\") - 1);
 }
 
+<ST_IN_SCRIPTING>"parent::$"{LABEL}"::"("get"|"set") {
+	RETURN_TOKEN_WITH_STR(T_PARENT_PROPERTY_HOOK_NAME, 0);
+}
```

Comment:
Why is this done with a lexer hack rather than rerouting it in zend_compile_static_call()?
i.e. checking for `class_ast->kind == ZEND_AST_STATIC_PROP`.

Is the goal being able to distinguish `(parent::$foo)::get()`?

Reply:
I honestly had not even considered it. The parenthesizing could be achieved by flagging the AST too, so no, that wasn't the deciding factor. But I do suspect that the compiler hack might not actually end up being simpler.

Reply:
As implemented, this is going to have unusual behavior wrt whitespace and comments, i.e. `parent::$foo::get()` and `parent::$foo ::get()` will not be the same thing (but both potentially valid...)

Reply:
I wasn't asking from a simplicity perspective, but what Nikita says - correctness with respect to whitespaces and comments and such. And I'd rather keep the lexer minimal with hacks needed.

Reply:
Ok. I pushed a new commit (https://github.com/php/php-src/pull/13455/commits/1d08b27d0f4512fb64e42f71ca7f144cf1fb190b). Please check if that works for you.

Reply:
LGTM :-)